### ISSUE TITLE
Selected day issue after upgrading to 1.1290.0

### ISSUE THREAD

**Issue by Spharian** [wix/react-native-calendars#2064]
## Description

Selected day is buggy after upgrading to 1.1290.0: every similar days on the agenda are also selected (I select the day 19, every 19 of every months are considered as selected also). See the screenshot below.

## Expected Behavior

Only the selected day should be taken into account for `selectedDayBackgroundColor` and `selectedDayTextColor` in the theme properties.

## Environment

```
react-native-calendars@1.1290.0
react-native@0.68.2

Device/emulator/simulator & OS version: iPhone 13 Pro - iOS 15.5
```

## Screenshots

<img width="308" alt="Screenshot 2022-10-20 at 12 19 33" src="https://user-images.githubusercontent.com/2619388/196923346-b3368bf2-b705-4e51-be56-19b8ff0b9614.png">

**Comment by macasfaj** [comment]
This version breaks a lot of things. backgroundColor in theme is ignored. I reverted to 1.1284.0 and it works like a charm

**Comment by riansco14** [comment]
@macasfaj is right, its works for me, amazing that they reported it even today and I had to use this library today.

yarn add react-native-calendars@1.1284.0

**Comment by dazx606** [comment]
I have the same issue

**Comment by luisbaldissera** [comment]
Same here! I have the impression this [getState](https://github.com/wix/react-native-calendars/blob/master/src/calendar/index.tsx#L208) is either returning `'selected'` when it shouldn't, or it is not being called at all due to some rendering behavior when changing the month and keeping the value of the selected date in another month.

**Comment by chunghn** [comment]
Thank you. Downgrading to 1.1284.0 fixed a lot of issues.

**Comment by Spharian** [comment]
Fixed in [1.1293.0](https://github.com/wix/react-native-calendars/releases/tag/1.1293.0)
